docs/architecture/ADR-003-borrow-assets.md

```diff
@@ -0,0 +1,121 @@
+# ADR 003: Borrowing assets using uToken collateral
+
+## Changelog
+
+- September 27, 2021: Initial Draft (@toteki)
+
+## Status
+
+Proposed
+
+## Context
+
+One of the base functions of the Umee universal capital facility is to allow users to borrow allowed asset types, using their own uTokens (obtained normally, by depositing assets) as collateral.
+
+## Alternative Approaches
+
+It might be worth considering adding message types which do two or more of the steps below at once (e.g. deposit base asset, convert to uToken collateral, and receive loan all at once). The individual steps to need to be possible in isolation, though.
+
+## Decision
+
+The Cosmos `x/bank` module and the existing `umee/x/leverage` deposit features are prerequisites for these new capabilities.
+
+The flow of events is as follows:
+- Borrower has already used `umee/x/leverage` module to deposit base assets and receive uTokens
+- Borrower deposits uTokens into asset facility as collateral - facility remembers collateral position
+- Borrower requests to borrow assets from capital facility - facility checks request, disburses tokens if acceptable, and remembers borrow position
```

Comment:
These two items are technically part of the same "step" right? In other words, the user has some uToken and requests to borrow X amount of some Token using some or all of their uToken.

Reply:
Is there a check in the state machine that allows borrowing of an asset based on the total reserves loaned?

In compound this is referred to as "collateral factor, ranging from 0 to 1".

Reply:
Also, when depositing on Compound it shows you the max borrow amount. Are we able to do that here as well?

Reply:
(Replying as if post-discussion edits already made): Since borrowers no longer need to deposit uTokens when using them as collateral, the first step doesn't happen.

The borrowing limit calculation (which uses collateral factor set by governance) happens when allowing/rejecting borrows and when transferring uTokens (to prohibit undercollateralization). We could display it too by doing the same calculation on the frontend.